### PR TITLE
allow public access to connection, and delay connection to first use

### PR DESCRIPTION
**Pull request by bumperbox**
Allows public access to the connection objects via getConnection()
Don't create connection in constructor, create it on first use

### CODE REVIEW

File: src/Cagartner/SQLAnywhereClient.php [hunk 1]
```diff
@@ -138,7 +122,7 @@ public function prepare($sql_string, $array=array())
 	 */
 	public function errorCode()
 	{
-		return sasql_errorcode( $this->connection ? $this->connection : null );
+		return sasql_errorcode( $this->getConnection() ? $this->getConnection() : null );
```
Comment (markcww) [inline src/Cagartner/SQLAnywhereClient.php]: just use this->getConnection (no need to test)

File: src/Cagartner/SQLAnywhereClient.php [hunk 2]
```diff
@@ -147,7 +131,7 @@ public function errorCode()
 	 */
 	public function errorInfo()
 	{
-		return sasql_error( $this->connection ? $this->connection : null );
+		return sasql_error( $this->getConnection() ? $this->getConnection() : null );
```
Comment (markcww) [inline src/Cagartner/SQLAnywhereClient.php]: same as above

File: src/Cagartner/SQLAnywhereClient.php [hunk 3]
```diff
@@ -170,16 +154,44 @@ public function commit()
...
 
 	public function __destruct(){
-	    return sasql_commit( $this->connection );
+	    return sasql_commit( $this->getConnection() );
   	}
 
   	// UNSUPPORTED PUBLIC METHODS
   	public function quote($data='')
 	{
 		return true;
```
Comment (markcww) [inline src/Cagartner/SQLAnywhereClient.php]: why doesn't the original code quote the data? Is this used?

File: src/Cagartner/SQLAnywhereClient.php [hunk 4]
```diff
@@ -39,37 +39,21 @@ class SQLAnywhereClient
...
+     * Create connection sybase
+     * @param string $dns String connection for sybase
+     * @param boolean $persistent Define connection for persistent
+     */
+    public function __construct($dns, $autocommit = true, $persistent = false) {
+        $this->dns = $dns;
+        $this->persistent = $persistent;
+        $this->autocommit = $autocommit;
+
+        if(!function_exists('sasql_connect')) {
```
Comment (markcww) [inline src/Cagartner/SQLAnywhereClient.php:48]: probably not necessary as this will fail anyway on first attempt to connect (but doesn't hurt)

**Review by markcww — APPROVED** [review]
LGTM (after minor changes in error functions)

File: src/Cagartner/SQLAnywhereClient.php (hunk 1 above)
Comment (bumperbox) [reply src/Cagartner/SQLAnywhereClient.php]: good point, i just did cut and paste replacements, i will fix that up

File: src/Cagartner/SQLAnywhereClient.php (hunk 3 above)
Comment (bumperbox) [reply src/Cagartner/SQLAnywhereClient.php]: not sure on the original code, i don't agree with all the design decisions, i fixed some when i converted to laravel 5.1 version, i will set up a new branch for that

File: src/Cagartner/SQLAnywhereClient.php (hunk 4 above)
Comment (bumperbox) [reply src/Cagartner/SQLAnywhereClient.php:48]: I left it in, cause its nice to have a human readable error message that is straight to the point

**Pull request merged**